src/Controls/src/BindingSourceGen/BindingCodeWriter.cs

```diff
@@ -40,7 +40,20 @@ private static bool ShouldUseSetter(BindingMode mode)
 		}
 		""";
 
-	private static string GenerateBindingCode(string bindingMethodBody) => $$"""
+	private static string GenerateUnsafeFieldAccessor(string fieldName, string memberType, string containingType, uint id) => $$"""
+		[UnsafeAccessor(UnsafeAccessorKind.Field, Name = "{{fieldName}}")]
+		private static extern ref {{memberType}} GetUnsafeField{{id}}{{fieldName}}({{containingType}} source);
+		""";
+
+	private static string GenerateUnsafePropertyAccessors(string propertyName, string memberType, string containingType, uint id) => $$"""
+		[UnsafeAccessor(UnsafeAccessorKind.Method, Name = "get_{{propertyName}}")]
+		private static extern {{memberType}} GetUnsafeProperty{{id}}{{propertyName}}({{containingType}} source);
+
+		[UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_{{propertyName}}")]
+		private static extern void SetUnsafeProperty{{id}}{{propertyName}}({{containingType}} source, {{memberType}} value);
```

Comment:
What will happen when the property doesn't have a setter? I would expect the compilation of the generated code to fail in that case. Let's cover that with a unit test + ideally let's not generate the unsafe property setter at all when there isn't a setter.

Reply:
The compilation will not fail. `UnsafeAccessor` can be used to modify a value of readonly fields / properties.

Reply:
But when the property doesn't have a setter, there won't be any `set_{{propertyName}}` method 🤔 would that only crash at runtime if the unsafe setter method is called?

Reply:
Oh sorry. I was fixated on fields. It will crush on runtime I believe: https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMA6AJWmAEsBbAUzIGEB7OgBxoBsGwBlPgDcaAYwYBnANx5ZuPKgDMJdEQAirVngDeeIvqIcwNIQENgDEpmIAVBgA9gRbQHMGwKQF883+X6UqRMzOegYBqAAsRACyABQAlCG4BkS+vgrKGFYA7Dqh+gDaAKpQEqYAZgwAgqLiEhKsYLElZZU1dQ1gANI0UHBk0e4AFqxwMEQAcqaMRAC8RABEbsAA+naOC/EAugqYAGxEDhZgUFbEAOLuLRUMAApgrBx8wACe68CxGqxEDRBg4vEZH5iqUbu1JJ1mqC2rUIY0en0BsNRuMpjN5gsJO41kdNjt8PtDo4+KdIkRBMBrpV7o9nm8jp9ND9WH9xONSMQzNwIAxAfkiPyMKp+bpkikDGYwEQ4HMiFAGAB3dSaBJA8UGClUu4PJ5gV7vWJjRamUybNXqqwATliCwAGmo1Gb+WkgA===